src/algorand/AlgorandEngine.ts

```diff
@@ -701,6 +701,22 @@ export class AlgorandEngine extends CurrencyEngine<
         rawTx.amount.toString(),
         recipient
       )
+
+      // Check if the recipient has activated/opted-in the asset
+      if (edgeTransaction.tokenId != null) {
```

Comment:
I don't think this block could ever run. The recipient is only defined for "pay" transaction types which are ALGO sends and it's undefined for "axfer" token sends. 

You'll also need to be sure this doesn't break token activation transactions which is a token send to yourself and the asset wouldn't be present in the account info response.